lib/iex/test/iex/interaction_test.exs

```diff
@@ -180,5 +180,14 @@ defmodule IEx.InteractionTest do
       File.rm("dot-iex-1")
       File.rm("dot-iex")
     end
+
+    test "malformed .iex" do
+      File.write!("dot-iex", "malformed")
+
+      assert capture_iex("malformed", [], dot_iex_path: "dot-iex") =~
+               "** (CompileError) iex:1: undefined function malformed/0"
```

Comment:
I think this should be `dot-iex` :)

```suggestion
               "** (CompileError) dot-iex:1: undefined function malformed/0"
```


Besides, I would use a different input for the `capture_iex` function so it's clear the error comes from the "dot iex" file.